File: core/src/main/java/heroes/journey/entities/Buff.java
package heroes.journey.entities;

import static heroes.journey.registries.Registries.BuffManager;

import heroes.journey.registries.Registrable;
import lombok.Getter;

@Getter
public class Buff extends Registrable {

    private final int turnsBuffLasts;
    private final int timesBuffCanBeUsed;

    public Buff(String id, String name, int turnsBuffLasts, int timesBuffCanBeUsed) {
        super(id, name);
        this.turnsBuffLasts = turnsBuffLasts;
        this.timesBuffCanBeUsed = timesBuffCanBeUsed;
    }

    public Buff(String id, String name) {
        this(id, name, -1, -1);
    }

    public Buff register() {
        return BuffManager.register(this);
    }
}

File: core/src/main/java/heroes/journey/entities/Buff.kt
package heroes.journey.entities

import heroes.journey.entities.tagging.Attributes
import heroes.journey.modlib.misc.IBuff
import heroes.journey.modlib.registries.Registrable
import heroes.journey.mods.Registries
import lombok.Getter

@Getter
class Buff(
    id: String,
    override val turnsBuffLasts: Int,
    override val attributes: Attributes
) :
    Registrable(id), IBuff {
    override fun register(): Buff {
        return Registries.BuffManager.register(this)
    }
}
